### PR TITLE
2024-08-12-preview API skeleton

### PR DESCRIPTION
**Pull request by kimorris27**
### Which issue this PR addresses:

https://issues.redhat.com/browse/ARO-4380

### What this PR does / why we need it:

Adds a new API skeleton in preparation for WIMI-related API work.

I followed the steps given at https://msazure.visualstudio.com/AzureRedHatOpenShift/_wiki/wikis/ARO.wiki/240029/API-Release-Process?anchor=detailed-steps-for-building-the-new-api-version (first few steps, stopping in front of "Make Changes to the New API").

### Test plan for issue:

I locally updated `pkg/util/cluster` to use the newly added preview API and then created and deleted a dev cluster to validate that I had set up the new API version properly.

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: pkg/api/v20240812preview/openshiftcluster.go
Comment (tsatam) [inline pkg/api/v20240812preview/openshiftcluster.go:85]: extremely minor pedantic nit: this needs to be changed to `Canceled` in order to comply with ARM Rest Specs requirements. We're working on backporting this change to the 20231122 API as well.

File: pkg/api/v20240812preview/openshiftcluster.go
Comment (tony-schndr) [inline pkg/api/v20240812preview/openshiftcluster.go:85]: Not sure what the preference is here, but in v20231122 I updated the comment and variable names to 'Canceled' as well.

**Comment by tony-schndr** [comment]
Hey Kipp, take a look at my backport PR https://github.com/Azure/ARO-RP/pull/3420 I just opened.  Besides my comment above, I backported x-ms-enum header additions (these were upstream but never made it to ARO-RP) and missing `WorkerProfilesStatus` in the api examples.

File: pkg/api/v20240812preview/openshiftcluster.go
Comment (tsatam) [reply pkg/api/v20240812preview/openshiftcluster.go:85]: We should ensure the same approach is taken on both APIs - in the small chance we have to cut a new stable API release off of 20231122 while 20230812preview is still active, we'd end up having to merge diffs between both APIs, just as we did on 20231122 itself. Keeping things as close together as possible will help make that process less painful.

Comment (kimorris27) [reply pkg/api/v20240812preview/openshiftcluster.go:85]: Oops, this was an oversight and wasn't an intentional choice. I'll fix it. 🙂

**Comment by kimorris27** [comment]
> Hey Kipp, take a look at my backport PR #3420 I just opened. Besides my comment above, I backported x-ms-enum header additions (these were upstream but never made it to ARO-RP) and missing `WorkerProfilesStatus` in the api examples.

Just read over it. I'm thinking that to avoid confusion and conflicts, we can consider my PR blocked until yours gets merged so that I can rebase to include your changes here.

**Comment by SudoBrendan** [comment]
hey @kimorris27 - just a heads-up that the REST Specs backport has just merged.

File: pkg/api/v20240812preview/openshiftcluster.go
Comment (SudoBrendan) [reply pkg/api/v20240812preview/openshiftcluster.go:85]: Hey - unsure what happened but this `const` name got reverted.

**Review by SudoBrendan — CHANGES_REQUESTED** [review]
Just a small diff between API versions - need `ProvisioningStateCanceled`

File: pkg/api/v20240812preview/openshiftcluster.go
Comment (kimorris27) [reply pkg/api/v20240812preview/openshiftcluster.go]: Should be fixed now!

**Review by SudoBrendan — APPROVED** [review]

**Review by cadenmarchese — APPROVED** [review]
lgtm!

**Comment by kimorris27** [comment]
CI was stuck and there was no E2E on the PR at all for some reason, so I just started both of them myself.

**Comment by cadenmarchese** [comment]
[As discussed](https://redhat-internal.slack.com/archives/C02ULBRS68M/p1709226939091489), we are going to proceed with merging this as e2e signal isn't a requirement and all other checks are green.

**Pull request merged**
